documentation/APIs/core-models.md

```diff
@@ -0,0 +1,581 @@
+  * [Ed25519 Signature](#ed25519-signature)
+
+## Block
+The following table describes the serialization of a Block:
+
+<table>
+  <tr>
+    <th>Name</th>
+    <th>Type</th>
+    <th>Description</th>
+  </tr>
+  <tr>
+    <td>Protocol Version</td>
+    <td>uint8</td>
+    <td>This field denotes what protocol rules apply to the block. 
+</td>
+  </tr>
+  <tr>
+    <td>Network ID</td>
+    <td>uint64</td>
+    <td>Network identifier. Usually, it will be set to the first 8 bytes of the BLAKE2b-256 hash of the concatenation of the network type and the protocol version string.</td>
+  </tr>
+  <tr>
+    <td>Strong Parents Count</td>
+    <td>uint8</td>
+    <td>The number of blocks that are strongly directly approved.</td>
+  </tr>
+  <tr>
+    <td valign="top">Strong Parents <code>anyOf</code></td>
+    <td colspan="2">
+      <details>
+        <summary>Parent</summary>
+        <blockquote>
+          References another directly approved block.
+        </blockquote>
+        <table>
+          <tr>
+            <th>Name</th>
+            <th>Type</th>
+            <th>Description</th>
+          </tr>
+          <tr>
+            <td>Block ID</td>
+            <td>ByteArray[40]</td>
+            <td>The Block ID of the strong parent.</td>
+          </tr>
+        </table>
+      </details>
+    </td>
+  </tr>
+  <tr>
+  <tr>
+    <td>Weak Parents Count</td>
+    <td>uint8</td>
+    <td>The number of blocks that are weakly directly approved.</td>
+  </tr>
+  <tr>
+    <td valign="top">Weak Parents <code>anyOf</code></td>
+    <td colspan="2">
+      <details>
+        <summary>Parent</summary>
+        <blockquote>
+          References another directly approved block.
+        </blockquote>
+        <table>
+          <tr>
+            <th>Name</th>
+            <th>Type</th>
+            <th>Description</th>
+          </tr>
+          <tr>
+            <td>Block ID</td>
+            <td>ByteArray[40]</td>
+            <td>The Block ID of the parent.</td>
+          </tr>
+        </table>
+      </details>
+    </td>
+  </tr>
+  <tr>
+  <tr>
+    <td>Shallow Like Parents Count</td>
+    <td>uint8</td>
+    <td>The number of blocks that are directly referenced to adjust opinion.</td>
+  </tr>
+  <tr>
+    <td valign="top">Shallow Like Parents <code>anyOf</code></td>
+    <td colspan="2">
+      <details>
+        <summary>Parent</summary>
+        <table>
+          <tr>
+            <th>Name</th>
+            <th>Type</th>
+            <th>Description</th>
+          </tr>
+          <tr>
+            <td>Block ID</td>
+            <td>ByteArray[40]</td>
+            <td>The Block ID of the parent.</td>
+          </tr>
+        </table>
+      </details>
+    </td>
+  </tr>
+  <tr>
+    <td>Issuer ID</td>
+    <td>ByteArray[32]</td>
+    <td>The issuer identifier</td>
+  </tr>
+  <tr>
+    <td>Issuing Time</td>
+    <td>int64</td>
+    <td>The time the block is issued.</td>
+  </tr>
+  <tr>
+    <td>Slot Commitment Length</td>
```

Comment:
What is this length for? A `Slot Commitment` size doesn't look dynamic.

Reply:
@jkrvivian I see that this was closed without answering or addressing it, can you let me know what's the reason for this field please?

Reply:
Yeah, I removed this field, it's a fixed-size and not an optional structure, so the structure is serialized anyway.
I also double-check with the bytes from serix, we don't need the length.